### DISCUSSION TITLE
Trial ends only after audio finished + participant responded  (audio_slider_response)

### DISCUSSION THREAD

**Discussion by kassyec** [jspsych/jsPsych#2044] [Q&A] [answered]
Hi there, 

I am using the audio slider plug-in. I want people to listen to the audio clip and then respond to the question with the slider. Unfortunately, the only options in this plug-in are: response ends trial and trial duration. 

All of my audio clips have different trial lengths. Right now I have it as response ends trial, which means that people have been going through the experiments very fast and not hearing the whole audio clip. 

I am confused as to how to get the continue button to pop up, or only allow the button to be pressed once the audio is finished playing. 

I was reading some other discussions about minimum time to read instructions and others for audio files, but I was confused about how to edit the plugins when I am using cognition.run with a built in library. 

Thanks so much! I would a[[reciate any help with this issue I am having.

**Comment by GEJ1** [comment, marked as answer]
Hi @kassyec ,

Have you tried with _response_allowed_while_playing parameter_?

 ```javascript 
response_allowed_while_playing: false  
 ```
From [documentation](https://www.jspsych.org/plugins/jspsych-audio-slider-response/):

> If true, then responses are allowed while the audio is playing. If false, then the audio must finish playing before the slider is enabled and the trial can end via the next button click. Once the audio has played all the way through, the slider is enabled and a response is allowed (including while the audio is being re-played via on-screen playback controls).

Best,

Gustavo.
